### PR TITLE
Fix: Non object rule options merge (fixes #3179)

### LINKED ISSUE #3179: Rule Merge Issue
.eslintrc
`"no-restricted-modules": [1, "nconf", "underscore"]`

public/js/.eslintrc
`"no-restricted-modules": [2, "requirejs"]`

console.logging the options inside of no-restricted-modules.js
```
[ 'nconf', 'underscore' ]
[ 'nconf', 'underscore' ]
[ 'nconf', 'underscore' ]
[ 'requirejs', 'underscore' ]
[ 'requirejs', 'underscore' ]
```

Expected:
```
[ 'nconf', 'underscore' ]
[ 'nconf', 'underscore' ]
[ 'nconf', 'underscore' ]
[ 'underscore' ]
[ 'underscore' ]
[ 'underscore' ]
````

### PR DESCRIPTION
**Pull request by gyandeeps**

### CODE REVIEW

File: tests/lib/util.js [hunk 1]
```diff
@@ -158,6 +158,38 @@ describe("util", function() {
...
+                { rules: { "no-mixed-requires": [1, "nconf", "underscore"] } },
+                { rules: { "no-mixed-requires": [2, "requirejs"] } }
+            ];
+
+            var result = util.mergeConfigs(code[0], code[1]);
+
+            assert.strictEqual(result.rules["no-mixed-requires"][0], 2);
+            assert.strictEqual(result.rules["no-mixed-requires"][1], "requirejs");
+            assert.isUndefined(result.rules["no-mixed-requires"][2]);
+            assert.deepEqual(code[0], { rules: { "no-mixed-requires": [1, "nconf", "underscore"] }});
```
Comment (xjamundx) [inline tests/lib/util.js]: unclear on the need for these assertions. are you just proving the originals were unmodified?

Comment (gyandeeps) [inline tests/lib/util.js]: correct. This used to fail when we used to do shallow copy.

**Comment by xjamundx** [comment]
:+1:

File: tests/lib/util.js [hunk 2]
```diff
@@ -158,6 +158,38 @@ describe("util", function() {
             assert.deepEqual(code[1], { rules: { "no-mixed-requires": [1, {"err": ["error", "e"]}] }});
         });
 
+        it("should combine configs and override rules options without array or object", function() {
+
+            var code = [
```
Comment (nzakas) [inline tests/lib/util.js]: config?

Comment (gyandeeps) [inline tests/lib/util.js]: I will make this change in whole file as the whole file has code in it.

**Comment by gyandeeps** [comment]
Done.

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**